models/users.js

```diff
@@ -4,24 +4,47 @@ class Users {
   }
 
   add(fakeName,realName) {
+    var name_check = this.check(fakeName, realName);
+    if (!name_check.success) { 
+      return name_check;
+    }
+    
     var new_user = {
       fakeName,
       realName,
       id: this.generateID()
     }
     this.list.push(new_user);
+    return {success: true, user: new_user}
   }
 
   generateID() {
     let ids = [];
-    this.list.forEach((element) => {
-      ids.push(element.id)
+    this.list.forEach((user) => {
+      ids.push(user.id)
     })
+    
     if(ids.length == 0) {
       return "1";
     }
     return (Math.max(...ids) + 1).toString();
   }
+
+  check(fakeName, realName) {
+    var success = true;
+    var reason = null;
+    this.list.forEach((user) => {
+      if (user.fakeName == fakeName) {
+        success = false
+        reason = "fakeName is already taken"
+      }
+      if (user.realName == realName) {
+        success = false
+        reason = "This real name is already taken. Maybe add your last name?"
+      }
```

Comment:
Maybe make these display on screen instead of a console log?

Reply:
Nice work guys

Reply:
Yes, will put error messages on display! :)